File: lib/ismd-engine/AbstractAuthController.php
<?php
/**
 * Родительский класс для контроллеров, которым необходимо,
 *   чтобы пользователь был авторизован
 *
 * @author ismd
 */

abstract class AbstractAuthController extends AbstractController {

    public function __construct($registry) {
        parent::__construct($registry);

        if (!isset($this->_session->user)) {
            header('Location: /');
            die;
        }
    }
}

File: server/src/main/java/game/server/controllers/common/AbstractAuthController.java
package game.server.controllers.common;

import game.Online;
import game.server.request.Request;
import game.user.User;

/**
 * @author ismd
 */
abstract public class AbstractAuthController extends AbstractController {

    @Override
    public boolean init(Request request) {
        User user = Online.users.get(request.getWs());

        if (null != user && null != user.getCurrentCharacter()) {
            return true;
        }

        String controller = request.getController();
        String action = request.getAction();

        return "Character".equals(controller) && ("create".equals(action) || "set".equals(action));
    }
}
